docs/DeveloperGuide.adoc

```diff
@@ -793,22 +837,20 @@ public static void saveDataToFile(Path file, XmlSerializableUser user) throws Fi
 }
 ----
 
-===== Implementation of the save feature
+
+---
 Given below is an example usage scenario and how the save mechanism behaves:
 
-The user launches the application for the first time and the `currentUser` will at this point be `null`.
-Issuing the save command at this point will throw an error message indicating to the user that they have not registered an account or they are currently not logged in.
+1. A student will first log into their account and add their preferred modules.
 
-In order to utilize the save function, the user must perform one of the following options:
+2. Upon issuing the command `save sp/userdata.xml`, the save command will call `Model#saveUserFile(...)` which then raises `SaveUserChangedEvent`. This `SaveUserChangedEvent` will be handled by `StorageManager`.
 
-* The user executes `register user/demo ...` to register a new `Student` account.
-The `save` command will call `Model#saveUserFile(...)`. +
+3. `StorageManager` then utilizes `XmlUserStorage#saveUser(...)` which saves the file to the file path specified by the user.
 
-* The user executes `login user/demo pass/#Qwerty1 userData/userdata.xml` to log into account.
-The `save` command will call `Model#saveUserFile(...)`. +
 
-[NOTE]
-A new `Student` would be initialized and automatically set as the `currentUser`, enabling the user to perform save command. +
+[TIP]
+A new registered `Student` would be initialized and automatically logged in which enabling the student to perform save command. +
```

Comment:
Minor grammatical errors:
_"which enabling"_ should be _"which enables"_
_"perform save"_ should be _"perform the save"_